app/src/main/java/org/mozilla/focus/session/ui/SessionViewHolder.kt

```diff
@@ -59,10 +58,7 @@ class SessionViewHolder internal constructor(private val fragment: SessionsSheet
     }
 
     private fun updateUrl(session: Session) {
-        val shortUrl = UrlUtils.stripCommonSubdomains(
-                UrlUtils.stripScheme(session.url.value))
-
-        textView.text = shortUrl.beautifyUrl()
+        textView.text = session.url.value.beautifyUrl()
```

Comment:
I wonder if we should make `beautifyUrl` a method on `Uri` instead:
1. it'd cause red flags when you pass an already stripped string into a Uri constructor
1. `BeautifyUrl` throws for error cases and then immediately converts to Uri (sounds like a Uri method :P).
1. It's more type-safe, by convention, than using String-encoded Urls

But the trade-off is we're not always passing around URLs as Uris so we could end up creating a lot of wasted objects.

Reply:
Yeah, that's what I was wondering too when I wrote this code initially. Back then I decided against adding the method to Uri because I didn't want to be forced to do the conversion everywhere I want to use the method. We are basically never using Uri objects and maybe it is just an implementation detail here (to extract parts of the URL easily). There's no reason why we couldn't remove the Uri conversion from beautifyUrl() and replace it with a different implementation. If it wouldn't require a lot of work I'd finally like to use our own URL class. This way we could finally remove all the thousand helper method and just attach them to our class. :)

Reply:
> If it wouldn't require a lot of work I'd finally like to use our own URL class. This way we could finally remove all the thousand helper method and just attach them to our class. :)

I really like this idea of making any urls we bring into the app type-safe by wrapping them in a URL type, rather than keeping them as Strings. With extension functions, this doesn't necessarily need to be our own URL class but then we maybe lose a little flexibility? I guess I haven't thought about it enough.